### PR TITLE
Aerohive captive portal

### PR DESCRIPTION
**Pull request by julsemaan**
Adds support for the aerohive captive portal
At the same time adds support for switch based web form authentication

For review and merge.

### CODE REVIEW

File: lib/pf/Switch/AeroHIVE/AP_http.pm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        $logger->info("Returning ACCEPT");
+        return [$RADIUS::RLM_MODULE_OK, %$radius_reply_ref];
+    }
+
+}
+
+sub getAcceptForm {
+    my ( $self, $mac , $destination_url) = @_;
+    my $logger = Log::Log4perl::get_logger( ref($self) );
+    $logger->info("MAC ADDRESS OF ZE ZEVISE $mac");
```
Comment (fdurand) [inline lib/pf/Switch/AeroHIVE/AP_http.pm]: Z´interesting to have ze zevise

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 2]
```diff
@@ -2680,6 +2680,38 @@ Caching and Roaming
...
+This option is currently broken but will be needed so it redirects the user at the end of the authentication.
+* Go in the section * Captive Web Portal Success Page Settings* and check *Redirect to the initially requested page* at the bottom.
+
+////
+
+[float]
+External portal SSID
+++++++++++++++++++++
+
+Again in the HVM, go to *Configuration -> SSIDs*, and create a new SSID with the following:
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Might tell what HVM is before using that acronym..

File: lib/pf/Switch/AeroHIVE/AP_http.pm [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    my ($self, $vlan, $mac, $port, $connection_type, $user_name, $ssid, $wasInline, $user_role) = @_;
+    my $logger = Log::Log4perl::get_logger( ref($self) );
+
+    my $radius_reply_ref = {};
+
+    my $node = node_view($mac);
+
+    my $violation = pf::violation::violation_view_top($mac);
+    # if user is unregistered or is in violation then we reject him to show him the captive portal 
+    if( $node->{status} eq $pf::node::STATUS_UNREGISTERED || defined($violation) ){
```
Comment (extrafu) [inline lib/pf/Switch/AeroHIVE/AP_http.pm]: if( -> if (  ?

I'm not sure we glue together braces/parenthesis in other parts of the code.

File: lib/pf/Switch/AeroHIVE/AP_http.pm (hunk 1 above)
Comment (extrafu) [inline lib/pf/Switch/AeroHIVE/AP_http.pm]: Zam mit -> Damn it?

File: lib/pf/Switch/AeroHIVE/AP_http.pm [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    return $html_form;
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2013 Inverse inc.
```
Comment (extrafu) [inline lib/pf/Switch/AeroHIVE/AP_http.pm]: 2014

File: lib/pf/Switch/AeroHIVE/AP_http.pm [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    return ($$req->param('Calling-Station-Id'),$$req->param('ssid'),$$req->param('STA-IP'),$$req->param('destination_url'),$$req->param('url'),"200");
+
+}
+
+=item returnRadiusAccessAccept
+
+Prepares the RADIUS Access-Accept reponse for the network device.
+
+Overiding the default implementation for the external captive portal
```
Comment (extrafu) [inline lib/pf/Switch/AeroHIVE/AP_http.pm]: Overiding -> Overriding

**Comment by julsemaan** [comment]
Closing to merge the Xirrus one with it since both are using a similar type of registration

**Pull request closed without merging**